radixdlt/src/main/java/com/radixdlt/consensus/sync/SyncManager.java

```diff
@@ -0,0 +1,150 @@
+package com.radixdlt.consensus.sync;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.LongConsumer;
+import java.util.function.LongSupplier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.radixdlt.middleware2.CommittedAtom;
+import com.radixdlt.utils.ThreadFactories;
+
+public class SyncManager {
+
+	private static final int MAX_REQUESTS_TO_SEND = 20;
+	private final Consumer<CommittedAtom> atomProcessor;
+	private final LongSupplier versionProvider;
+	private final int batchSize;
+	private final int maxAtomsQueueSize;
+	private final long patience;
+
+	private final TreeSet<CommittedAtom> commitedAtoms = new TreeSet<>((a1, a2) -> Long
+			.compare(a1.getVertexMetadata().getStateVersion(), a2.getVertexMetadata().getStateVersion()));
```

Comment:
Comparator could be simplified to `Comparator.comparingLong(a -> a.getVertexMetadata().getStateVersion())`

Reply:
Done